### ISSUE TITLE
[Bug] AADSTS900438: Refresh token value is missing.

### ISSUE THREAD

**Issue by jay-ng** [AzureAD/microsoft-authentication-library-for-dotnet#4176]
**Logs and network traces**
The app is on the user's device. I'm still working on a "how" to get the log. 

**Which version of MSAL.NET are you using?**
MSAL.NET 4.54.1

**Platform**
Xamarin iOS

**What authentication flow has the issue?**
* Desktop / Mobile
    * [x] Interactive (broker)
    * [ ] Interactive (embedded or system browser)
    * [ ] Integrated Windows Authentication
    * [ ] Username / Password
    * [ ] Device code (browserless)
* Web app 
    * [ ] Authorization code
    * [ ] On-Behalf-Of
* Daemon app 
    * [ ] Client credentials (service-to-service calls)

Other?

**Is this a new or existing app?**
New implementation of MSAL on a currently published app. 

**Repro**

```csharp
string[] Scopes = { "User.Read" };
var settings = new GlobalSettings();
var clientId = settings.GetClientId();
var builder = PublicClientApplicationBuilder.Create(clientId);
builder = builder.WithIosKeychainSecurityGroup("com.microsoft.adalcache");
builder = builder.WithRedirectUri("<redirect-uri>"); // msauth.<bundle-id>://auth
builder = builder.WithBroker();
Client = builder.Build();

AuthenticationResult authResult = null;
IEnumerable<IAccount> accounts = await Client.GetAccountsAsync().ConfigureAwait(false);
try
{
    IAccount firstAccount = accounts.FirstOrDefault();
    authResult = await Client.AcquireTokenSilent(Scopes, firstAccount)
                          .ExecuteAsync()
                          .ConfigureAwait(false);
}
catch (MsalUiRequiredException)
{
    try
    {
        var builder = Client.AcquireTokenInteractive(Scopes)
            .WithParentActivityOrWindow(Application.RootViewController);
        authResult = await builder.ExecuteAsync().ConfigureAwait(false);
    }
    catch (Exception ex2)
    {
        throw new UnauthorizedAccessException("Acquire token interactive failed. See exception message for details:" + ex2.Message);
    }
}
```

**Actual behavior**
Behavior on my device: 
Successful authentication.

Behavior on user's device:
During the AcquireTokenInteractive flow, the broker (Microsoft Authenticator) will redirect back to the mobile app. That's when "AADSTS900438: Refresh token value is missing." is thrown in the inner catch block.

**Expected behavior**
AcquireTokenInteractive flow successfully authenticate with broker.

**Possible solution**

**Additional context / logs / screenshots / links to code**
Correlation ID: 066ece71-e744-4278-89b9-85e9d1e8e152

**Comment by gladjohn** [comment]
@jay-ng is it happening only for a single user? are there any MSAL logs? looks more like a Microsoft Authenticator broker issue to me

**Comment by gladjohn** [comment]
> > @jay-ng is it happening only for a single user? are there any MSAL (verbose) logs? looks more like a Microsoft Authenticator broker issue to me. Can you also collect [broker logs](https://learn.microsoft.com/en-us/troubleshoot/azure/active-directory/send-microsoft-authenticator-logs) if possible?
> 
> Please stand by. I'm working on collecting those.

Thanks once you have the broker logs we can pass it on to the IOS Broker team. Is this a cross cloud request?

**Comment by jay-ng** [comment]
> 
> Thanks once you have the broker logs we can pass it on to the IOS Broker team. Is this a cross cloud request?

@gladjohn, This is a US cloud request. Here are the extra details that were gathered:

- Incident ID: 67KMKUWG. Are there any other steps I need to do to get the log from MS Authenticator?
- On another user device:
  - Once brokered by MS Authenticator, the first error she was getting was: "Broker response nonce does not match the request nonce sent by MSAL.Net"
  - Second try, back to the usual error: "Refresh token value is missing"

I suspect the first error also showed up for the previous user, but only once.
Please let me know if this help.

**Comment by bgavrilMS** [comment]
I believe this was fixed in MSAL 4.57 and up by https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/4395
